frontend/src/metabase/admin/settings/components/widgets/CustomGeoJSONWidget.tsx

```diff
@@ -0,0 +1,490 @@
+import cx from "classnames";
+import { type ChangeEvent, memo, useCallback, useState } from "react";
+import { t } from "ttag";
+
+import { useLazyLoadGeoJSONQuery } from "metabase/api/geojson";
+import { useAdminSetting } from "metabase/api/utils";
+import { ConfirmModal } from "metabase/components/ConfirmModal";
+import { LoadingAndErrorWrapper } from "metabase/components/LoadingAndErrorWrapper";
+import Modal from "metabase/components/Modal";
+import { Ellipsified } from "metabase/core/components/Ellipsified";
+import Select, { Option } from "metabase/core/components/Select";
+import AdminS from "metabase/css/admin.module.css";
+import ButtonsS from "metabase/css/components/buttons.module.css";
+import CS from "metabase/css/core/index.css";
+import { uuid } from "metabase/lib/uuid";
+import { Button } from "metabase/ui";
+import LeafletChoropleth from "metabase/visualizations/components/LeafletChoropleth";
+import type {
+  CustomGeoJSONMap,
+  CustomGeoJSONSetting,
+  GeoJSONData,
+} from "metabase-types/api";
+
+import { SettingHeader } from "../SettingHeader";
+
+export const CustomGeoJSONWidget = () => {
+  const [map, setMap] = useState<CustomGeoJSONMap | undefined>();
+  const [originalMap, setOriginalMap] = useState<
+    CustomGeoJSONMap | undefined
+  >();
+  const [currentId, setCurrentId] = useState<string | undefined>();
+  const [
+    triggerLoadGeoJSON,
+    { data: geoJson, error: geoJsonError, isLoading: geoJsonLoading },
```

Comment:
```suggestion
    { data: geoJson, error: geoJsonError, isFetching: geoJsonLoading },
```

I think we need to use is `isFetching` prop here instead of `isLoading` - if i load one geojson, it will show the cached one while loading another one instead of the loading spinner

![slow-geojson](https://github.com/user-attachments/assets/04eca05b-e052-4105-a557-137793b06fb3)